data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconStateHandler.java

```diff
@@ -13,26 +13,62 @@
 
 package tech.pegasys.artemis.beaconrestapi.beaconhandlers;
 
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.apache.tuweni.bytes.Bytes32;
-import tech.pegasys.artemis.beaconrestapi.handlerinterfaces.BeaconRestApiHandler;
+import tech.pegasys.artemis.beaconrestapi.schema.BeaconStateResponse;
+import tech.pegasys.artemis.datastructures.state.BeaconState;
+import tech.pegasys.artemis.provider.JsonProvider;
 import tech.pegasys.artemis.storage.ChainStorageClient;
 
-public class BeaconStateHandler implements BeaconRestApiHandler {
-
+public class BeaconStateHandler implements Handler {
+  public static final String ROUTE = "/beacon/state/";
+  private final Logger LOG = LogManager.getLogger();
   private final ChainStorageClient client;
 
   public BeaconStateHandler(ChainStorageClient client) {
     this.client = client;
   }
 
-  @Override
-  public String getPath() {
-    return "/beacon/state";
+  private BeaconState queryByRootHash(String root) {
+    Bytes32 root32 = Bytes32.fromHexString(root);
+    return client.getStore().getBlockState(root32);
   }
 
+  @OpenApi(
+      path = BeaconStateHandler.ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get the beacon chain state that matches the specified tree hash root.",
+      tags = {"beacon"},
+      queryParams = {
+        @OpenApiParam(name = "root", description = "Tree hash root to query (Bytes32)")
+      },
+      description =
+          "Request that the node return a beacon chain state that matches the provided criteria.",
+      responses = {
+        @OpenApiResponse(
+            status = "200",
+            content = @OpenApiContent(from = BeaconStateResponse.class)),
+        @OpenApiResponse(status = "404")
+      })
   @Override
-  public Object handleRequest(RequestParams params) {
-    Bytes32 root = Bytes32.fromHexString(params.getQueryParam("root"));
-    return client.getStore() != null ? client.getStore().getBlockState(root) : null;
+  public void handle(Context ctx) throws Exception {
+    String rootParam = ctx.queryParam("root");
+    BeaconState result = queryByRootHash(rootParam);
+    if (result == null) {
+      LOG.debug("Block root {} not found", rootParam);
```

Comment:
Suggest trace level not debug because it's essentially just recording the flow control that was followed.

Reply:
ok